helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -519,13 +519,20 @@ public static InstanceConfig composeInstanceConfig(String instanceName) {
   }
 
   /**
-   * Checks whether or not the cluster is in management mode.
+   * Checks whether or not the cluster is in management mode. It checks pause signal
```

Comment:
Question, is management mode defined as Pause + cluster freeze, or also with Maintenance mode? There have been some changes and I want to confirm.

Reply:
Currently we only make it for pause & freeze. In the future when we move M mode to management pipeline, we should also check M mode. A TODO is already added to this API.

Reply:
Sounds good.